### PR TITLE
Playwright: add mobile support to pages used by `seo_preview` spec.

### LINKED ISSUE #53803: Playwright: add mobile support to components and pages
**Task Details**
Current implementation of the `SidebarComponnt` only supports viewport sizes that constantly show the sidebar. There are lots of other places that have a similar issue, where the narrow viewport will hide or condense some elements.

Look at refactoring and adding proper mobile support for these components so that Playwright mobile tests can be turned on for trunk.

**Related to**

### PR DESCRIPTION
**Pull request by worldomonation**
#### Changes proposed in this Pull Request

This PR implements mobile viewport support for MarketingPage, used by the `seo_preview` spec.

#### Testing instructions

TeamCity
- ensure tests pass
- if muted tests are present, check if the mutes are caused by the `seo_preview` spec.

Related to #53803

### CODE REVIEW

File: packages/calypso-e2e/src/lib/pages/marketing-page.ts [hunk 1]
```diff
@@ -11,13 +11,14 @@ import { Page } from 'playwright';
...
-	navtabsList: '.section-nav-tabs__list',
+	navTabs: 'div.section-nav-tabs',
+	navTabsDropdownOptions: '.select-dropdown__option',
 
 	// Traffic tab
 	websiteMetaTextArea: '#advanced_seo_front_page_description',
 	seoPreviewButton: '.seo-settings__preview-button',
 	seoPreviewPane: '.web-preview.is-seo',
-	seoPreviewPaneCloseButton: '.web-preview .web-preview__close',
+	seoPreviewPaneCloseButton: '.web-preview__close',
```
Comment (worldomonation) [inline packages/calypso-e2e/src/lib/pages/marketing-page.ts:21]: This selector was too specific and also different between `mobile` and `desktop` viewports.

File: packages/calypso-e2e/src/lib/pages/marketing-page.ts [hunk 2]
```diff
@@ -32,16 +33,7 @@ export class MarketingPage extends BaseContainer {
...
-	}
-
-	/**
-	 * Post-initialization steps when creating an instance of this object.
-	 *
-	 * @returns {Promise<void>} No return value.
-	 */
-	async _postInit(): Promise< void > {
-		await this.page.waitForSelector( selectors.navtabsList );
+		super( page );
```
Comment (worldomonation) [inline packages/calypso-e2e/src/lib/pages/marketing-page.ts:36]: Removed the default selector wait.

File: packages/calypso-e2e/src/lib/pages/marketing-page.ts [hunk 3]
```diff
@@ -11,13 +11,14 @@ import { Page } from 'playwright';
 
 const selectors = {
 	content: '#primary',
-	navtabsList: '.section-nav-tabs__list',
+	navTabs: 'div.section-nav-tabs',
+	navTabsDropdownOptions: '.select-dropdown__option',
```
Comment (worldomonation) [inline packages/calypso-e2e/src/lib/pages/marketing-page.ts]: Our navtabs become a non-standard dropdown when the viewport is small enough.

This also means that in `clickTabItem`, it isn't possible to simply use Playwright's built-in `selectOption` call.

Comment (WunderBart) [inline packages/calypso-e2e/src/lib/pages/marketing-page.ts]: nit:
```suggestion
	navTabsDropdownOption: '.select-dropdown__option',
```

File: packages/calypso-e2e/src/lib/pages/marketing-page.ts [hunk 4]
```diff
@@ -51,9 +43,18 @@ export class MarketingPage extends BaseContainer {
...
+		const navTabs = await this.page.waitForSelector( selectors.navTabs );
+		const isDropdown = await navTabs
+			.getAttribute( 'class' )
+			.then( ( value ) => value?.includes( 'is-dropdown' ) );
 		const sanitizedName = toTitleCase( [ name ] );
 
-		await this.page.click( `text=${ sanitizedName }` );
+		if ( isDropdown ) {
+			await navTabs.click();
+			await this.page.click( `${ selectors.navTabsDropdownOptions } >> text=${ name }` );
```
Comment (WunderBart) [inline packages/calypso-e2e/src/lib/pages/marketing-page.ts]: following up on previous comment: 
```suggestion
			await this.page.click( `${ selectors.navTabsDropdownOption } >> text=${ name }` );
```

**Review by WunderBart — APPROVED** [review]
Just one nitpick there! 👍

File: packages/calypso-e2e/src/lib/pages/marketing-page.ts (hunk 3 above)
Comment (worldomonation) [reply packages/calypso-e2e/src/lib/pages/marketing-page.ts]: I think I used a similar convention elsewhere in the codebase, where I used plural ending for selectors that will match against multiple items (eg. gallery items, select dropdown options, etc).

Would you say it is worth codifying to use singular naming where possible, and perhaps go back and remove these plural endings (lower priority item)?

Comment (WunderBart) [reply packages/calypso-e2e/src/lib/pages/marketing-page.ts]: Yeah, let's keep the singular naming. For one it's closer to the original selector. I think plurals are generally used for containers/wrappers (`.nav-tabs`, `.comments`).

**Pull request merged**